docs/proposals/202005_version_documentation.md

```diff
@@ -0,0 +1,164 @@
+            |- master
+            |- version 0.13.0
+            |- version 0.12.2
+            |- other-folder-for-other-releases
+```
+_NOTE: `tmp` directory is not committed, just temporarily built. The current version of docs lives in the `master` folder_
+
+
+#### 3. Building a versioning plugin
+
+Creating a plugin that can automate these processes would save us a lot of development time and stress. This approach promises to be useful when it comes to versioning different release in the Thanos website.
+
+
+##### Workflow
+
+1. The developer makes all the necessary edits in `/docs` on master.
+2. The developer proceeds by committing a new release (i.e Release 0.x).
+3. CI run some `make web` or `make web-serve`.
+4. Before anything, generates the docs and places them in a `versioned` tmp folder.
+5. the rest of the web command is executed.
+
+_NOTE: generated docs are not committed, just temporarily built._
+
+## FAQ
+
+This section consists of some important questions and answers that are frequently asked by users.
+
+##### How do you specify what part of the docs you want to version without cloning every file? docs.yaml
+
+We hope to have a single, flexible configuration file (`docs.yaml`) that will help the developer in specifying what part of the docs they need without cloning all the docs. We could have this as a single file on master. The config file will look like this:
+
+```versioned:
+   default:
+   - docs/components/*.md
+   - docs/storage.md
+
+  overrides:
+     release-0.12:
+     - docs/storage2221241.md
+```
+##### Alternative path
+
+We could have this file on master, so the current `make web` will
+
+1. Parse `docs.yaml` on master and decide.
+1. Go to that particular release (`i.e release-0.10`)
+1. Is there docs.yaml?
+    * Yes - parse and decide
+    * No - use docs.yaml from master
+
+The design of docs.yaml will look like this:
+
+```versioned:
+   - docs/components/*.md
+   - docs/storage2221241.md
+```
+##### What would the plugin look like?
+
+CLI (`make generate-versioned-docs`)
+
+##### What is the plan in terms of plugin placement?
+
+We hope to start in the Thanos project and then move the project outside to a separate repository.
+
+##### How we do fixes to release docs for older release, e.g. v0.12, work?
+
+We will edit the particular release (release-0.12) branch and add a commit. The then tool fetches the latest commit on this branch and uses it to generate the docs. We expect to fetch docs for minor releases without breaking them with patches. We encourage immutability across all our release tags in Thanos.
+
+##### How does the tool discover the releases for fixes?
+
+With a regular expression. Instead of the developer manually checking out the individual release branches, the tool handles it for them by using a regex to select and clone valid release branches. The tool (config file) will have a `release-branch-regex` field. For Thanos, the regular expression would be something like `release-(.*)`.
```

Comment:
```suggestion
With a regular expression. Instead of the developer manually checking out the individual release branches, the tool handles it for them by using a regex to select and clone valid release branches. The config file for the tool will have a `release-branch-regex` field. For Thanos, the regular expression would be something like `release-(.*)`.
```